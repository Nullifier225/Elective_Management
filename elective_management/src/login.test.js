import React from 'react';
import { shallow } from 'enzyme';
import Login from './component/login';
import {mount} from 'enzyme';
    describe('Login component tests', ()=> {
        const wrapper = shallow(<Login />);

        it('should have a btn component', ()=> {

            //There should be only one button
            expect(wrapper.find('button#Button')).toHaveLength(1);


            //Button should have matching text
            expect(JSON.parse(JSON.stringify(wrapper.find('button').text()))).toEqual("Sign in");
        });

        it('should have the Amrita Logo', ()=> {

            //There should be only one button
            expect(wrapper.find('img#Amrita')).toHaveLength(1);

        });

        it('should have input for email and password', ()=> {
            //Email and password input field should be present
            expect(wrapper.find('input#email')).toHaveLength(1);
            expect(wrapper.find('input#pwd')).toHaveLength(1);
        });


        it('should have an empty email and password state var', ()=> {
            expect(wrapper.state('email')).toEqual('');
            expect(wrapper.state('password')).toEqual('');
        });

        //UNIT TESTING 

        test('submitting form is sucessful', async () => {
            
            
            wrapper.find('input[type="password"]').simulate('change', {target: {name: 'password', value: 'login'}});
            wrapper.find('input[type="email"]').simulate('change', {target: {name: 'email', value: 'cb.en.u4cse18042@cb.students.amrita.edu'}});
            expect(wrapper.state('password')).toEqual('login');
            expect(wrapper.state('email')).toEqual('cb.en.u4cse18042@cb.students.amrita.edu');
            expect(wrapper.find("form").simulate("submit", { preventDefault() {} }))

            //console.log((wrapper.find('form#myForm').simulate('submit')));
            //wrapper.find('form#myForm').simulate('submit')
            //expect(submit_signup).toHaveBeenCalledTimes(1);
               
            

               
});


    ;


   
  
 
  


    });

















    