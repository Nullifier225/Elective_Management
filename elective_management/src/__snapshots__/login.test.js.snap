// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component tests should test Header component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          class="img1"
          src="https://image3.mouthshut.com/images/Restaurant/Photo/-73020_62102.jpg"
        />,
        <div
          class="out"
          style={
            Object {
              "height": "auto",
            }
          }
        >
          <img
            className="rounded mx-auto d-block"
            id="Amrita"
            src="index.png"
            style={
              Object {
                "height": "25%",
                "width": "25%",
              }
            }
          />
          <br />
          <form
            id="myForm"
            onSubmit={[Function]}
          >
            <h3
              class="heading1"
              className="text-center font-weight-bold font-size:1.5em"
            >
              Login
            </h3>
            <br />
            <div
              className="form-group"
            >
              <label>
                Email ID
              </label>
              <input
                className="form-control"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Enter Email ID"
                type="email"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label>
                Password
              </label>
              <input
                className="form-control"
                id="pwd"
                name="password"
                onChange={[Function]}
                placeholder="Enter password"
                type="password"
                value=""
              />
            </div>
            <button
              className="btn btn-info btn-lg btn-block"
              id="Button"
              type="submit"
              value="Sign in"
            >
              Sign In 
              <ForwardRef
                class="hover:bg-gray-100"
                color="white"
                size={30}
              />
            </button>
            <br />
          </form>
          <br />
          <div
            id="data"
          />
        </div>,
      ],
      "class": "out1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "class": "img1",
          "src": "https://image3.mouthshut.com/images/Restaurant/Photo/-73020_62102.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="rounded mx-auto d-block"
              id="Amrita"
              src="index.png"
              style={
                Object {
                  "height": "25%",
                  "width": "25%",
                }
              }
            />,
            <br />,
            <form
              id="myForm"
              onSubmit={[Function]}
            >
              <h3
                class="heading1"
                className="text-center font-weight-bold font-size:1.5em"
              >
                Login
              </h3>
              <br />
              <div
                className="form-group"
              >
                <label>
                  Email ID
                </label>
                <input
                  className="form-control"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Enter Email ID"
                  type="email"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label>
                  Password
                </label>
                <input
                  className="form-control"
                  id="pwd"
                  name="password"
                  onChange={[Function]}
                  placeholder="Enter password"
                  type="password"
                  value=""
                />
              </div>
              <button
                className="btn btn-info btn-lg btn-block"
                id="Button"
                type="submit"
                value="Sign in"
              >
                Sign In 
                <ForwardRef
                  class="hover:bg-gray-100"
                  color="white"
                  size={30}
                />
              </button>
              <br />
            </form>,
            <br />,
            <div
              id="data"
            />,
          ],
          "class": "out",
          "style": Object {
            "height": "auto",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "rounded mx-auto d-block",
              "id": "Amrita",
              "src": "index.png",
              "style": Object {
                "height": "25%",
                "width": "25%",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  class="heading1"
                  className="text-center font-weight-bold font-size:1.5em"
                >
                  Login
                </h3>,
                <br />,
                <div
                  className="form-group"
                >
                  <label>
                    Email ID
                  </label>
                  <input
                    className="form-control"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter Email ID"
                    type="email"
                    value=""
                  />
                </div>,
                <div
                  className="form-group"
                >
                  <label>
                    Password
                  </label>
                  <input
                    className="form-control"
                    id="pwd"
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter password"
                    type="password"
                    value=""
                  />
                </div>,
                <button
                  className="btn btn-info btn-lg btn-block"
                  id="Button"
                  type="submit"
                  value="Sign in"
                >
                  Sign In 
                  <ForwardRef
                    class="hover:bg-gray-100"
                    color="white"
                    size={30}
                  />
                </button>,
                <br />,
              ],
              "id": "myForm",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                  "class": "heading1",
                  "className": "text-center font-weight-bold font-size:1.5em",
                },
                "ref": null,
                "rendered": "Login",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label>
                      Email ID
                    </label>,
                    <input
                      className="form-control"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter Email ID"
                      type="email"
                      value=""
                    />,
                  ],
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Email ID",
                    },
                    "ref": null,
                    "rendered": "Email ID",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "id": "email",
                      "name": "email",
                      "onChange": [Function],
                      "placeholder": "Enter Email ID",
                      "type": "email",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label>
                      Password
                    </label>,
                    <input
                      className="form-control"
                      id="pwd"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter password"
                      type="password"
                      value=""
                    />,
                  ],
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Password",
                    },
                    "ref": null,
                    "rendered": "Password",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "id": "pwd",
                      "name": "password",
                      "onChange": [Function],
                      "placeholder": "Enter password",
                      "type": "password",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Sign In ",
                    <ForwardRef
                      class="hover:bg-gray-100"
                      color="white"
                      size={30}
                    />,
                  ],
                  "className": "btn btn-info btn-lg btn-block",
                  "id": "Button",
                  "type": "submit",
                  "value": "Sign in",
                },
                "ref": null,
                "rendered": Array [
                  "Sign In ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "class": "hover:bg-gray-100",
                      "color": "white",
                      "size": 30,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "defaultProps": Object {
                        "color": "currentColor",
                        "size": "1em",
                      },
                      "propTypes": Object {
                        "color": [Function],
                        "size": [Function],
                      },
                      "render": [Function],
                    },
                  },
                ],
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "data",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            class="img1"
            src="https://image3.mouthshut.com/images/Restaurant/Photo/-73020_62102.jpg"
          />,
          <div
            class="out"
            style={
              Object {
                "height": "auto",
              }
            }
          >
            <img
              className="rounded mx-auto d-block"
              id="Amrita"
              src="index.png"
              style={
                Object {
                  "height": "25%",
                  "width": "25%",
                }
              }
            />
            <br />
            <form
              id="myForm"
              onSubmit={[Function]}
            >
              <h3
                class="heading1"
                className="text-center font-weight-bold font-size:1.5em"
              >
                Login
              </h3>
              <br />
              <div
                className="form-group"
              >
                <label>
                  Email ID
                </label>
                <input
                  className="form-control"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Enter Email ID"
                  type="email"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <label>
                  Password
                </label>
                <input
                  className="form-control"
                  id="pwd"
                  name="password"
                  onChange={[Function]}
                  placeholder="Enter password"
                  type="password"
                  value=""
                />
              </div>
              <button
                className="btn btn-info btn-lg btn-block"
                id="Button"
                type="submit"
                value="Sign in"
              >
                Sign In 
                <ForwardRef
                  class="hover:bg-gray-100"
                  color="white"
                  size={30}
                />
              </button>
              <br />
            </form>
            <br />
            <div
              id="data"
            />
          </div>,
        ],
        "class": "out1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "class": "img1",
            "src": "https://image3.mouthshut.com/images/Restaurant/Photo/-73020_62102.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                className="rounded mx-auto d-block"
                id="Amrita"
                src="index.png"
                style={
                  Object {
                    "height": "25%",
                    "width": "25%",
                  }
                }
              />,
              <br />,
              <form
                id="myForm"
                onSubmit={[Function]}
              >
                <h3
                  class="heading1"
                  className="text-center font-weight-bold font-size:1.5em"
                >
                  Login
                </h3>
                <br />
                <div
                  className="form-group"
                >
                  <label>
                    Email ID
                  </label>
                  <input
                    className="form-control"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter Email ID"
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label>
                    Password
                  </label>
                  <input
                    className="form-control"
                    id="pwd"
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter password"
                    type="password"
                    value=""
                  />
                </div>
                <button
                  className="btn btn-info btn-lg btn-block"
                  id="Button"
                  type="submit"
                  value="Sign in"
                >
                  Sign In 
                  <ForwardRef
                    class="hover:bg-gray-100"
                    color="white"
                    size={30}
                  />
                </button>
                <br />
              </form>,
              <br />,
              <div
                id="data"
              />,
            ],
            "class": "out",
            "style": Object {
              "height": "auto",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "rounded mx-auto d-block",
                "id": "Amrita",
                "src": "index.png",
                "style": Object {
                  "height": "25%",
                  "width": "25%",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    class="heading1"
                    className="text-center font-weight-bold font-size:1.5em"
                  >
                    Login
                  </h3>,
                  <br />,
                  <div
                    className="form-group"
                  >
                    <label>
                      Email ID
                    </label>
                    <input
                      className="form-control"
                      id="email"
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter Email ID"
                      type="email"
                      value=""
                    />
                  </div>,
                  <div
                    className="form-group"
                  >
                    <label>
                      Password
                    </label>
                    <input
                      className="form-control"
                      id="pwd"
                      name="password"
                      onChange={[Function]}
                      placeholder="Enter password"
                      type="password"
                      value=""
                    />
                  </div>,
                  <button
                    className="btn btn-info btn-lg btn-block"
                    id="Button"
                    type="submit"
                    value="Sign in"
                  >
                    Sign In 
                    <ForwardRef
                      class="hover:bg-gray-100"
                      color="white"
                      size={30}
                    />
                  </button>,
                  <br />,
                ],
                "id": "myForm",
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Login",
                    "class": "heading1",
                    "className": "text-center font-weight-bold font-size:1.5em",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <label>
                        Email ID
                      </label>,
                      <input
                        className="form-control"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        placeholder="Enter Email ID"
                        type="email"
                        value=""
                      />,
                    ],
                    "className": "form-group",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Email ID",
                      },
                      "ref": null,
                      "rendered": "Email ID",
                      "type": "label",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "form-control",
                        "id": "email",
                        "name": "email",
                        "onChange": [Function],
                        "placeholder": "Enter Email ID",
                        "type": "email",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <label>
                        Password
                      </label>,
                      <input
                        className="form-control"
                        id="pwd"
                        name="password"
                        onChange={[Function]}
                        placeholder="Enter password"
                        type="password"
                        value=""
                      />,
                    ],
                    "className": "form-group",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Password",
                      },
                      "ref": null,
                      "rendered": "Password",
                      "type": "label",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "form-control",
                        "id": "pwd",
                        "name": "password",
                        "onChange": [Function],
                        "placeholder": "Enter password",
                        "type": "password",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Sign In ",
                      <ForwardRef
                        class="hover:bg-gray-100"
                        color="white"
                        size={30}
                      />,
                    ],
                    "className": "btn btn-info btn-lg btn-block",
                    "id": "Button",
                    "type": "submit",
                    "value": "Sign in",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Sign In ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "class": "hover:bg-gray-100",
                        "color": "white",
                        "size": 30,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "defaultProps": Object {
                          "color": "currentColor",
                          "size": "1em",
                        },
                        "propTypes": Object {
                          "color": [Function],
                          "size": [Function],
                        },
                        "render": [Function],
                      },
                    },
                  ],
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
              ],
              "type": "form",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "data",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSeventeenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
